### ISSUE TITLE
'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by ljhOfGithub** [pyg-team/pytorch_geometric#4286]
### 🐛 Describe the bug

    >>> from deeprobust.graph.data import Dataset
    >>> from deeprobust.graph.defense import ChebNet
    >>> data = Dataset(root='/tmp/', name='cora')
    >>> adj, features, labels = data.adj, data.features, data.labels
    >>> idx_train, idx_val, idx_test = data.idx_train, data.idx_val, data.idx_test
    >>> cheby = ChebNet(nfeat=features.shape[1],
              nhid=16, num_hops=3,
              nclass=labels.max().item() + 1,
              dropout=0.5, device='cpu')
    >>> cheby = cheby.to('cpu')
    >>> pyg_data = Dpr2Pyg(data) # convert deeprobust dataset to pyg dataset
    >>> cheby.fit(pyg_data, patience=10, verbose=True) # train with earlystopping
#https://github.com/DSE-MSU/DeepRobust/blob/master/deeprobust/graph/defense/chebnet.py
#the pyg_data may be an object of Dpr2Pyg,and its len(data_list) == 1, according to the in_memory_dataset.py, the return value of slices is None,but when I run the last sentence 'cheby.fit(pyg_data, patience=10, verbose=True)',the bug report is ''NoneType' object is not subscriptable',I comment the    '     # if len(data_list) == 1:        #     return data_list[0], None'  sentences, and it works.

### Environment

* PyG version:latest
* PyTorch version:1.10.0
* OS:windows 10 19044.1586
* Python version:3.8.5
* CUDA/cuDNN version:cpu
* How you installed PyTorch and PyG (`conda`, `pip`, source):conda
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Two fixes are needed:
* https://github.com/DSE-MSU/DeepRobust/blob/master/deeprobust/graph/defense/chebnet.py#L116 needs to be replaced with:
   ```python
   self.device = self.conv1.lins[0].weight.device
   ```
* https://github.com/DSE-MSU/DeepRobust/blob/master/deeprobust/graph/data/pyg_dataset.py#L77 needs
   ```python
   if self.slices is None:
        return self.data
   ```
   before the `for` loop.

Linking to https://github.com/DSE-MSU/DeepRobust/issues/104 as well.
